lib/api/collection_config.rb

```diff
@@ -0,0 +1,120 @@
+module Api
+  class CollectionConfig
+    def self.names_for_feature(product_feature_name)
+      names_for_features[product_feature_name]
+    end
+
+    def self.name_for_klass(resource_klass)
+      Settings.collections.detect do |_, spec|
+        spec[:klass] && spec[:klass].constantize == resource_klass
+      end.try(:first)
+    end
+
+    def self.what_refers_to_feature(product_feature_name)
+      referenced_identifiers[product_feature_name]
+    end
+
+    def self.collections_with_description
+      Settings.collections.each_with_object({}) do |(collection, cspec), result|
+        result[collection] = cspec[:description] if cspec[:options].include?(:collection)
+      end
+    end
+
+    def self.find(collection_name)
+      config = Settings.collections[collection_name.to_sym]
+      new(config) if config
```

Comment:
I prefer [] and .vms.  Consistent usage with Settings, which is what these are and not really an AR model.

And yes, it is for efficiency, no need to allocate an object for each time we reference a setting.